api/src/main/java/org/eclipse/microprofile/openapi/annotations/OpenAPIDefinition.java

```diff
@@ -67,11 +68,25 @@
 
     /**
      * A declaration of which security mechanisms can be used across the API.
-     *
-     * @return the array of servers used for this API
+     * <p>
+     * Adding a {@code SecurityRequirement} to this array is equivalent to adding a {@code SecurityRequirementsSet} containing
+     * a single {@code SecurityRequirement} to {@link #securitySet()}.
+     * 
+     * @return the array of security requirements for this API
      */
     SecurityRequirement[] security() default {};
 
+    /**
+     * A declaration of which security mechanisms can be used across the API.
+     * <p>
+     * All of the security requirements within any one of the sets must be satisfied to authorize a request.
+     * <p>
+     * Including an empty set within this list indicates that the other requirements are optional.
+     * 
+     * @return the array of security requirement sets for this API
+     */
+    SecurityRequirementsSet[] securitySet() default {};
+
```

Comment:
Looking at it again, I think it might be more natural for this and the property in `CallbackOperation` to be plural, `securitySets`.